### ISSUE TITLE
Classic FM support?

### ISSUE THREAD

**Issue by klop422** [rg3/youtube-dl#17868]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.10.05*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.10.05**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

```
youtube-dl -v -x --audio-format mp3 https://www.classicfm.com/radio/aod/?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-x', '--audio-format', 'mp3', 'https://www.classicfm.com/radio/aod/?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2018.10.05
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.17134
[debug] exe versions: ffmpeg N-87257-g8e17cd2, ffprobe N-87257-g8e17cd2
[debug] Proxy map: {}
[generic] ?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043: Requesting header
WARNING: Falling back on generic information extractor.
[generic] ?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043: Downloading webpage
[generic] ?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043: Extracting information
ERROR: Unsupported URL: https://www.classicfm.com/radio/aod/?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp5xdqinp2\build\youtube_dl\YoutubeDL.py", line 792, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp5xdqinp2\build\youtube_dl\extractor\common.py", line 507, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp5xdqinp2\build\youtube_dl\extractor\generic.py", line 3298, in _real_extract
youtube_dl.utils.UnsupportedError: Unsupported URL: https://www.classicfm.com/radio/aod/?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single stream: https://www.classicfm.com/radio/aod/?episodeId=17fe8e48-2389-49ba-b4e2-bf835af59043
- Single stream (other example): https://www.classicfm.com/radio/aod/?episodeId=34bee1be-06d2-4ed1-a3e4-0ea32e3bca90

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

Simply put, could support be added for Classic FM streams? The website allows you to listen to the live radio shows again for up to seven days (i.e. there's no copyright infringement involved on their part)
Note, the link won't work unless you're in the UK (or give a UK Postcode).
I've given only links for single streams because I don't believe they do playlists or any other kind of link.

**Comment by klop422** [comment]
A bump, and also giving more example links because they expire after a week:
https://www.classicfm.com/radio/player/
Any link under the 'listen again' tab on that page should work as an example.

**Comment by condemnedmeat** [comment]
Classicfm shows are trivial to download. You don't need youtube-dl.

Once on the page featuring the on demand show you want to listen to, right click on the page and inspect. Search - `ctrl+f` - for "m4a" or "dax", this yields an address, mixed in with other info. The address looks like this: http://catchup.hosting.thisisdax.com/f7df68b4-4cde-48f6-a0d0-639516754012/1583cec9-d92b-43f3-8f87-142d19b6f00d.m4a

Just paste it into the address bar and it'll give you the audio.
